cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -820,6 +824,11 @@ private SearchResults searchByCustomIndex(SearchRequest request) throws BadReque
   }
 
   private void write(MetadataEntity metadataEntity, MetadataEntry entry, Set<Indexer> indexers) {
+    writeWithoutHistory(metadataEntity, entry, indexers);
+    writeHistory(metadataEntity);
+  }
+
+  private void writeWithoutHistory(MetadataEntity metadataEntity, MetadataEntry entry, Set<Indexer> indexers) {
```

Comment:
better name is writeValueWithIndexes

Reply:
This is named like that to show it does not write history and there is a separate method to write it.

Reply:
The name is wrong. Change it. The name should signify what it does not what it doesn't do. In later case the naming possibility is endless.

Reply:
fixed.